src/lib/containers/synapse_table_functions/SynapseTableCell.tsx

```diff
@@ -4,70 +4,39 @@ import React from 'react'
 import { PRODUCTION_ENDPOINT_CONFIG } from '../../utils/functions/getEndpoint'
 import { AUTHENTICATED_USERS } from '../../utils/SynapseConstants'
 import {
-  ColumnModel,
+  ColumnType,
   FileHandleAssociateType,
-  SelectColumn,
+  UserGroupHeader,
+  UserProfile,
 } from '../../utils/synapseTypes'
 import { EntityHeader } from '../../utils/synapseTypes/EntityHeader'
-import {
-  CardLink,
-  ColumnSpecifiedLink,
-  MarkdownLink,
-} from '../CardContainerLogic'
 import DirectDownload from '../DirectDownload'
 import EntityIdList from '../EntityIdList'
 import { EntityLink } from '../EntityLink'
-import { SynapseCardLabel } from '../GenericCard'
-import { Dictionary } from '../table/SynapseTable'
 import { NOT_SET_DISPLAY_VALUE } from '../table/SynapseTableConstants'
 import UserCard from '../UserCard'
 import { ElementWithTooltip } from '../widgets/ElementWithTooltip'
 
 type SynapseTableCellProps = {
-  entityColumnIndicies: number[]
-  userColumnIndicies: number[]
-  dateColumnIndicies: number[]
-  dateListColumnIndicies: number[]
-  booleanListColumnIndicies: number[]
-  fileHandleIdColumnIndicies: number[]
-  entityIdListColumnIndicies: number[]
-  otherListColumnIndicies: number[]
-  colIndex: number
+  columnType: ColumnType
   columnValue: string
   isBold: string
-  mapEntityIdToHeader: Dictionary<EntityHeader>
-  mapUserIdToHeader: Dictionary<any>
-  columnLinkConfig?: MarkdownLink | CardLink | ColumnSpecifiedLink
```

Comment:
We no longer need columnLinkConfig, selectColumns, columnModels, or rowData because the SynapseCardLink logic has been extracted from this component